src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/Tokenization.cs

```diff
@@ -0,0 +1,320 @@
+                if (complilerGeneratedCallNodes.Contains(call.Node.Id))
+                {
+                    continue;
+                }
+
+                int spanStart = call.Node.Head.Token.Span.Min;
+                int spanEnd = call.Node.Head.Token.Span.Lim; // span.Lim
+
+                // Close the Function token at the end of the call word instead of the right closing parenthesis.
+                Contracts.Assert(spanEnd >= spanStart, "The Span End value should be larger than the Span Start.");
+                tokens.Add(new TokenTextSpan(call.Node.Head.Name, spanStart, spanEnd, TokenType.Function, /* canHide */ false));
+
+                // Parse the call arguments delimiters
+                if (call.Node.Args.Delimiters != null)
+                {
+                    string delimiterName = null;
+                    foreach (var delimiter in call.Node.Args.Delimiters)
+                    {
+                        delimiterName = call.Node.Head.Name + "_delimeter";
+                        tokens.Add(new TokenTextSpan(delimiterName, delimiter.Span.Min, delimiter.Span.Lim, TokenType.Delimiter, /* canHide */ false));
+                    }
+                }
+            }
+
+            if (comments != null)
+            {
+                foreach (var cm in comments)
+                {
+                    span = new Span(cm.Span.Min, cm.Span.Lim);
+                    tokens.Add(new TokenTextSpan(TexlLexer.PunctuatorBlockComment, span.Min, span.Lim, TokenType.Comment, false));
+                }
+            }
+
+            foreach (var info in binding.GetControlKeywordInfos())
+            {
+                var textSpan = info.Node.GetCompleteSpan();
+                span = new Span(textSpan.Min, textSpan.Lim);
+                TokenType type = MapNodeKindToTokenType(info.Node.Kind);
+
+                // Add a token for the keyword.
+                tokens.Add(new TokenTextSpan(info.Name, span.Min, span.Lim, type, false));
+            }
+
+            // Binary Operators
+            ExtractTokensFromNodesList(tokens, binding.GetBinaryOperators(), TokenizerConstants.BinaryOp);
+
+            // Variadic Operators
+            ExtractTokensFromNodesList(tokens, binding.GetVariadicOperators(), TokenizerConstants.VariadicOp);
+
+            // Boolean Literals
+            ExtractTokensFromNodesList(tokens, binding.GetBooleanLiterals(), TokenizerConstants.BooleanLiteral);
+
+            // Numeric Literals
+            ExtractTokensFromNodesList(tokens, binding.GetNumericLiterals(), TokenizerConstants.NumericLiteral);
+
+            // Decimal Literal Nodes
+            ExtractTokensFromNodesList(tokens, binding.GetDecimalLiterals(), TokenizerConstants.DecimalLiteral);
+
+            // String Literals
+            ExtractTokensFromNodesList(tokens, binding.GetStringLiterals(), TokenizerConstants.StringLiteral);
+
+            // Unary Operators ie Currently just used for Percentages
```

Comment:
>Currently just used for Percentages [](http://example.com/codeflow?start=34&length=35)

Not unary negate or not?

Reply:
Yes, i copied over this comment as well but i did look into this now that you pointed out and we only add/track UnaryOpNode whose Token kind is TokKind.PercentSign in the Binder.cs
<img width="361" alt="image" src="https://user-images.githubusercontent.com/25670945/234394861-98378989-be88-45ee-a77a-47a55ed94d0a.png">
Since we only track UnaryOpNode with PercentSign, binding.GetUnaryOperators() returns nodes with Token.Kind == TokKind.PercentSign. Also added a quick test to confirm this